### ISSUE TITLE
Provide dataset as a stream

### ISSUE THREAD

**Issue by csmith1991** [csmith1991/d-sas7bdat#6]
SAS datasets can get quite large. I've seen 10s of gigabytes in single datasets. We should be able to handle this kind of volume. I'm not sure what the best way to provide test files for this is... suggestions?

**Comment by John-Colvin** [comment]
You could always memory map the file.

Accessing the data as a range would be ideal from a user standpoint. If you memory map then you get an array, which is itself a range. However, I don't know enough about the format to advise too definitely whether this would work out well.

**Comment by csmith1991** [comment]
True, I'll need to test performance with it. It's not uncommon that the files will not fit in memory. I've seen some as large as 50GB, and I'm sure those aren't even the biggest, so I really just need to do something.

**Comment by John-Colvin** [comment]
What exactly comprises a dataset? Is it one big table of data or could it be many different things all together?

**Comment by csmith1991** [comment]
One table with a fixed number of columns and rows in pages
